calyx-frontend/src/ast.rs

```diff
@@ -341,6 +343,54 @@ pub enum Control {
         /// Attributes
         attributes: Attributes,
     },
+    /// Represents sequential composition of static control statements.
+    StaticSeq {
+        /// List of `Control` statements to run in sequence.
+        /// If not all of these stmts are static, we should error out
+        stmts: Vec<Control>,
+        /// Attributes
+        attributes: Attributes,
+        /// Latency for the seq
+        latency: Option<u64>,
+    },
+    /// Represents sequential composition of static control statements.
+    StaticPar {
+        /// List of `Control` statements to run in sequence.
+        /// If not all of these stmts are static, we should error out
+        stmts: Vec<Control>,
+        /// Attributes
+        attributes: Attributes,
+        /// Latency for the seq
+        latency: Option<u64>,
```

Comment:
Interesting... the only question is, if someone _does_ annotate with latency of 0, do we want to have the latency as 0, then throw an error when it doesn't match the latency that we infer? If we use NonZerou64, it seems like it will just act as if the (incorrect) 0 latency annotation is not there-- which seems fine too. Ig it's a preference.

Reply:
We can probably do that in the parser itself using the `input.error` message and reject that program before it appears in the AST?